iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2174,6 +2176,49 @@ static STRING_HANDLE buildClientId(const char* device_id, const char* module_id)
     }
 }
 
+static void append_optional_connect_parameters(PMQTTTRANSPORT_HANDLE_DATA transport_data)
```

Comment:
>append_optional_connect_parameters [](start = 12, length = 34)

This function should return an error, since failirue to set DCM specifically should cause fatal errors.